### PR TITLE
Updates for dependent project usage

### PR DESCRIPTION
**Pull request by timkimadobe**
> [!NOTE]
> A lot of the lines are from the addition of the JSON comparison tool implementation in `XCTestCase+AnyCodableAsserts.swift` which was brought in from Edge extension implementation as-is (except for adding the `public` access modifier in front of the public APIs)

## Description

This PR updates the testing utilities to:
1. Allow access to test util classes' and extensions', public methods by updating to `public` access level
3. Updates framework dependency to include XCTest, for automatic linking in dependent projects

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [ ] I have signed the [Adobe Open Source CLA](https://opensource.adobe.com/cla.html).
- [ ] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

File: AEPTestUtils.podspec [hunk 1]
```diff
@@ -18,4 +18,5 @@ Pod::Spec.new do |s|
   s.dependency 'AEPCore', '>= 4.0.0'
 
   s.source_files = 'Sources/**/*.swift'
+  s.frameworks   = 'XCTest'
 end
```
Comment (emdobrin) [inline AEPTestUtils.podspec:22]: we should target both ios and tvos.deployment_target = '11.0' similar to the SDK

File: Sources/MockDataQueue.swift [hunk 2]
```diff
@@ -14,48 +14,48 @@
...
-public class MockDataQueue: DataQueue {
+open class MockDataQueue: DataQueue {
     // swiftlint:disable identifier_name
-    let queue = ThreadSafeArray<DataEntity>()
+    public let queue = ThreadSafeArray<DataEntity>()
 
     public init() {}
 
-    public func add(dataEntity: DataEntity) -> Bool {
+    open func add(dataEntity: DataEntity) -> Bool {
```
Comment (emdobrin) [inline Sources/MockDataQueue.swift:23]: I would not recommend keeping all these mocks and other utils open where possible, in order to reduce the amount of code overlap and simplify the debugging experience. My understanding is that if additional support or modifications are required, contributors would participate to the development of this repository, or for custom implementations a separate mock can be created implementing all the APIs from the defined protocol.

File: Sources/TestBase.swift [hunk 3]
```diff
@@ -11,52 +11,51 @@
...
+    public static func == (lhs: EventSpec, rhs: EventSpec) -> Bool {
         return lhs.source.lowercased() == rhs.source.lowercased() && lhs.type.lowercased() == rhs.type.lowercased()
     }
 
-    func hash(into hasher: inout Hasher) {
+    public func hash(into hasher: inout Hasher) {
         hasher.combine(type)
         hasher.combine(source)
     }
 }
```
Comment (emdobrin) [inline Sources/TestBase.swift:35]: might be good to move it in its own file now for discoverability

File: Sources/CountDownLatch.swift [hunk 4]
```diff
@@ -14,12 +14,12 @@ import Foundation
...
 
 /// CountDown latch to be used for asserts and expectations
-class CountDownLatch {
-    private let initialCount: Int32
-    private var currentCount: Int32
-    private let waitSemaphore = DispatchSemaphore(value: 0)
+open class CountDownLatch {
+    public let initialCount: Int32
+    public var currentCount: Int32
+    public let waitSemaphore = DispatchSemaphore(value: 0)
```
Comment (emdobrin) [inline Sources/CountDownLatch.swift:20]: I think the intention here was to keep these properties private to the class, and only expose them through the getters. Can we revert this change?

File: Sources/InstrumentedExtension.swift [hunk 5]
```diff
@@ -15,36 +15,36 @@ import AEPServices
...
 /// Instrumented extension that registers a wildcard listener for intercepting events in current session. Use it along with `TestBase`
-class InstrumentedExtension: NSObject, Extension {
-    private static let logTag = "InstrumentedExtension"
-    var name = "com.adobe.InstrumentedExtension"
-    var friendlyName = "InstrumentedExtension"
-    static var extensionVersion = "1.0.0"
-    var metadata: [String: String]?
-    var runtime: ExtensionRuntime
+open class InstrumentedExtension: NSObject, Extension {
+    public static let logTag = "InstrumentedExtension"
```
Comment (emdobrin) [inline Sources/InstrumentedExtension.swift:19]: private log tag, and are all these fields meant to be public?

File: Sources/NetworkRequestHelper.swift [hunk 6]
```diff
@@ -20,13 +20,13 @@ import XCTest
...
 ///    - ``RealNetworkService``
-class NetworkRequestHelper {
-    private var sentNetworkRequests: [TestableNetworkRequest: [NetworkRequest]] = [:]
+open class NetworkRequestHelper {
+    public var sentNetworkRequests: [TestableNetworkRequest: [NetworkRequest]] = [:]
     /// Matches sent `NetworkRequest`s with their corresponding `HttpConnection` responses.
-    private(set) var networkResponses: [TestableNetworkRequest: [HttpConnection]] = [:]
-    private var expectedNetworkRequests: [TestableNetworkRequest: CountDownLatch] = [:]
+    public private(set) var networkResponses: [TestableNetworkRequest: [HttpConnection]] = [:]
+    public var expectedNetworkRequests: [TestableNetworkRequest: CountDownLatch] = [:]
```
Comment (emdobrin) [inline Sources/NetworkRequestHelper.swift:27]: same comment here, we should keep these properties private since they are accessible through the APIs below

File: Sources/NetworkRequestHelper.swift [hunk 7]
```diff
@@ -203,22 +203,22 @@ class NetworkRequestHelper {
...
     ///   - timeout:how long should this method wait, in seconds; by default it waits up to 1 second
-    func wait(_ timeout: UInt32? = TestConstants.Defaults.WAIT_TIMEOUT) {
+    open func wait(_ timeout: UInt32? = TestConstants.Defaults.WAIT_TIMEOUT) {
         if let timeout = timeout {
             sleep(timeout)
         }
     }
 }
 
 extension URL {
```
Comment (emdobrin) [inline Sources/NetworkRequestHelper.swift:213]: might be good to extract this in its own file

**Review by timkimadobe — COMMENTED** [review]
Thanks for the review @emdobrin! I think the biggest question is the appropriate access level in: https://github.com/adobe/aepsdk-testutils-ios/pull/1#discussion_r1376963825

I will also work on moving some of the extensions etc into separate files as suggested!

**Review by timkimadobe — COMMENTED** [review]
Thanks for the guidance and feedback @emdobrin! Updated based on feedback - access levels should better reflect the design intent for the test utils

File: AEPTestUtils.podspec [hunk 8]
```diff
@@ -11,11 +11,15 @@ Pod::Spec.new do |s|
...
   s.source           = { :git => "https://github.com/adobe/aepsdk-testutils-ios.git", :tag => s.version.to_s }
-  s.ios.deployment_target = '10.0'
+  
+  s.ios.deployment_target = '11.0'
+  s.tvos.deployment_target = '11.0'
+
   s.swift_version = '5.1'
 
   s.pod_target_xcconfig = { 'BUILD_LIBRARY_FOR_DISTRIBUTION' => 'YES' }
   s.dependency 'AEPCore', '>= 4.0.0'
```
Comment (addb) [inline AEPTestUtils.podspec:21]: Should we add AEPServices as a dependency as well? I think Core depends on services so it should be downloaded nonetheless.

File: AEPTestUtils.podspec (hunk 1 above)
Comment (emdobrin) [reply AEPTestUtils.podspec:25]: Can you also check if this also matches the settings in xcode project for iOS_DEPLOYMENT_TARGET and tvOS_DEPLOYMENT_TARGET and Package.swift?

**Review by emdobrin — APPROVED** [review]
Looks good, left a small question
